nilearn/_utils/tests/test_niimg_conversions.py

```diff
@@ -383,17 +380,20 @@ def test_iter_check_niimgs(tmp_path):
     with pytest.raises(ValueError, match=no_file_matching % nofile_path):
         list(_iter_check_niimg(nofile_path))
 
+
+def test_iter_check_niimgs(tmp_path, img_4d_zeros_eye):
+    img_2_4d = [[img_4d_zeros_eye, img_4d_zeros_eye]]
+
     # Create a test file
     fd, filename = tempfile.mkstemp(
         prefix="nilearn_test", suffix=".nii", dir=str(tmp_path)
     )
     os.close(fd)
-    img_4d.to_filename(filename)
+    img_4d_zeros_eye.to_filename(filename)
```

Comment:
Is it possible to use the context manager ` testing.write_tmp_imgs` here like in some tests above?

Reply:
done

Reply:
OK it seems that this does not play well with testing on windows

Reply:
will revert for now